holoviews/element/raster.py

```diff
@@ -615,13 +615,14 @@ def dimension_values(self, dim, unique=False):
             d2_half_unit = (t - b)/dim2/2.
             d1lin = np.linspace(l+d1_half_unit, r-d1_half_unit, dim1)
             d2lin = np.linspace(b+d2_half_unit, t-d2_half_unit, dim2)
-            if unique:
-                return d2lin if dim_idx else d1lin
-            else:
+            if expanded:
                 Y, X = np.meshgrid(d2lin, d1lin)
                 return Y.flatten() if dim_idx else X.flatten()
+            else:
+                return d2lin if dim_idx else d1lin
         elif dim_idx == 2:
-            return np.flipud(self.data).T.flatten()
+            data = np.flipud(self.data).T
+            return data.flatten() if flat else data
```

Comment:
The values are the flattened, transposed ``flipud`` of the data array? This may well be correct but we will need to know why this is necessary and document it.